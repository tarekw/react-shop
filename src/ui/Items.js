import Container from "react-bootstrap/Container";
import Row from "react-bootstrap/Row";
import Card from "react-bootstrap/Card";

export const Items = ({ category, selectItem = (f) => f }) => (
  <Container fluid="sm" style={{ margin: 0 }}>
    <Row>
      {category.map((item, index) => (
        <Card
          style={{ width: "10rem" }}
          className="cardItem"
          onClick={() => selectItem(index)}
          key={item.id}
        >
          <Card.Img variant="top" src={item.image} />
          <Card.Body>
            <Card.Text>{item.title}</Card.Text>
          </Card.Body>
        </Card>
      ))}
    </Row>
  </Container>
);
