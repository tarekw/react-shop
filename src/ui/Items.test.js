import renderer from "react-test-renderer";
import { shallow } from "enzyme";

import Card from "react-bootstrap/Card";

import { Items } from "./Items";

describe("Items", () => {
  let category;
  let selectItem;

  beforeEach(() => {
    category = [
      {
        id: 9,
        title: "WD 2TB Elements Portable External Hard Drive - USB 3.0 ",
        price: 64,
        description: "description",
        category: "electronics",
        image: "https://fakestoreapi.com/img/61IBBVJvSDL._AC_SY879_.jpg",
        rating: {
          rate: 3.3,
          count: 203,
        },
      },
      {
        id: 10,
        title: "SanDisk SSD PLUS 1TB Internal SSD - SATA III 6 Gb/s",
        price: 109,
        description: "description",
        category: "electronics",
        image: "https://fakestoreapi.com/img/61U7T1koQqL._AC_SX679_.jpg",
        rating: {
          rate: 2.9,
          count: 470,
        },
      },
    ];
    selectItem = jest.fn();
  });

  it("should match snapshot", () => {
    const tree = renderer
      .create(<Items category={category} selectItem={selectItem} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should handle item selection", () => {
    const wrapper = shallow(<Items category={category} selectItem={selectItem} />);
    wrapper.find(Card).last().simulate("click")
    expect(selectItem).toHaveBeenCalledTimes(1);
    expect(selectItem).toHaveBeenCalledWith(1);
  });
});
