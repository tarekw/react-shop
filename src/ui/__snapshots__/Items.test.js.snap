// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Items should match snapshot 1`] = `
<div
  className="container-sm"
  style={
    Object {
      "margin": 0,
    }
  }
>
  <div
    className="row"
  >
    <div
      className="cardItem card"
      onClick={[Function]}
      style={
        Object {
          "width": "10rem",
        }
      }
    >
      <img
        className="card-img-top"
        src="https://fakestoreapi.com/img/61IBBVJvSDL._AC_SY879_.jpg"
      />
      <div
        className="card-body"
      >
        <p
          className="card-text"
        >
          WD 2TB Elements Portable External Hard Drive - USB 3.0 
        </p>
      </div>
    </div>
    <div
      className="cardItem card"
      onClick={[Function]}
      style={
        Object {
          "width": "10rem",
        }
      }
    >
      <img
        className="card-img-top"
        src="https://fakestoreapi.com/img/61U7T1koQqL._AC_SX679_.jpg"
      />
      <div
        className="card-body"
      >
        <p
          className="card-text"
        >
          SanDisk SSD PLUS 1TB Internal SSD - SATA III 6 Gb/s
        </p>
      </div>
    </div>
  </div>
</div>
`;
